### PR TITLE
refactor(build): clean and reorganize gulpfile

### PR DESCRIPTION
**Pull request by ncoden**
*not really useful, I know, but I had a little lack of motivation this weekend, and gulp interested me, so I did that.*

### Changes:
Global:
- Add comments/doc
- Reorganize into "config", "dependencies" and "tasks"

Dependencies:
- Remove unused dependencies
- Sort and group by purpose

Config:
- Standardize paths and file/directory names to paths when possible
- Sort and group by task
- Simplify and standardize options names

Tasks:
- Separate application and vendors process
- Sort by build order

Changes outside GulpFile:
- Move SCSS vendors to "/vendor"

### :warning: Require:
- [ ] https://github.com/CaliOpen/caliopen.web-client-ng/pull/52

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,83 +1,162 @@
...
+    .pipe(iconfontCss({
+      cssClass: iconfontConfig.iconsCssClass,
+      fontName: iconfontConfig.iconsFontName,
+      fontPath: config.clientCssFontsPath + '/' + iconfontConfig.iconsFontName + '/',
+      targetPath: '../../../' + config.buildDirectory + '/icons/' + iconfontConfig.iconsFontName + '.scss'
+    }))
+    .pipe(iconfont({
+      fontName: iconfontConfig.iconsFontName,
+      formats: ['ttf', 'eot', 'svg', 'woff', 'woff2'],
+      timestamp: runTimestamp, // recommended to get consistent builds when watching files
```
Comment (iamdey) [inline gulpfile.js]: in case of watch it should be better the timestamp is regenerated on runtime instead of gulp initialization

**Comment by iamdey** [comment]
it's ok for me. 
I had to clean manually the `dist` directory because of a fail. But I think it's not big deal:

```
[14:27:13] Using gulpfile ~/Workspace/caliopen/web-client-ng/gulpfile.js
[14:27:13] Starting 'build'...
[14:27:13] Starting 'clean'...
[14:27:13] Finished 'clean' after 19 ms
[14:27:13] Starting 'build:assets'...
[14:27:13] Starting 'build:assetsVendors'...
[14:27:13] Starting 'build:assetsIndex'...
[14:27:13] Starting 'build:assetsIcons'...
[14:27:13] Starting 'build:assetsFiles'...
[14:27:13] 'build:assetsVendors' errored after 80 ms
[14:27:13] Error: ENOENT: no such file or directory, chmod '/home/dey/Workspace/caliopen/web-client-ng/dist/fonts/bootstrap'
    at Error (native)
[14:27:13] 'build:assets' errored after 81 ms
[14:27:13] Error: ENOENT: no such file or directory, chmod '/home/dey/Workspace/caliopen/web-client-ng/dist/fonts/bootstrap'
    at Error (native)
[14:27:13] 'build' errored after 103 ms
[14:27:13] Error: ENOENT: no such file or directory, chmod '/home/dey/Workspace/caliopen/web-client-ng/dist/fonts/bootstrap'
    at Error (native)
[14:27:13] 'build:assetsFiles' errored after 45 ms
[14:27:13] Error: ENOENT: no such file or directory, chmod '/home/dey/Workspace/caliopen/web-client-ng/dist/fonts/Open_Sans'
    at Error (native)
[14:27:13] gulp-svgicons2svgfont: Font created
[14:27:13] Finished 'build:assetsIndex' after 731 ms
[14:27:13] Finished 'build:assetsIcons' after 736 ms
```

File: gulpfile.js (hunk 1 above)
Comment (ncoden) [inline gulpfile.js]: So... why merge ?

Comment (iamdey) [inline gulpfile.js]: not big deal, isn't it ?

**Pull request merged**